Tasks/PowerShellOnTargetMachines2/task.json

```diff
@@ -146,6 +146,18 @@
             "helpMarkDown": "Select the protocol to use for the WinRM connection with the machine(s). Default is HTTPS.",
             "groupName": "SessionOptions"
         },
+        {
+            "name": "AuthenticationMechanism",
+            "type": "pickList",
+            "label": "Authentication Mechanism",
+            "options": {
+                "Default": "Default",
+                "Credssp": "CredSSP"
```

Comment:
This needs enabling credssp as both client and server. We need to automate that as part of the task if this option is chosen. The role should also be removed once the task completes.

Reply:
For client, enabling credssp auth requires you to be running as local system ( or you need to provide admin credentials for agent machine). And admin credentials will also be required for enabling it on server side (the 2 credentials can be different). And that may not always be the case, as invoke-command can run scripts remotely if you are a member of the remote management users security group but not an administrator.